src/Microsoft.Health.Dicom.Api/Registration/DicomServerServiceCollectionExtensions.cs

```diff
@@ -105,6 +105,9 @@ public static IDicomServerBuilder AddBackgroundWorkers(this IDicomServerBuilder
 
             services.TryAddSingleton<RecyclableMemoryStreamManager>();
 
+            // Used by the BackgroundServiceHealthCheck
+            services.AddApplicationInsightsTelemetry();
```

Comment:
We do this already in the web project https://github.com/microsoft/dicom-server/blob/805968201c83be0a9cb841efdedb5a0bb272bf9d/src/Microsoft.Health.Dicom.Web/Startup.cs#L63

It feels like this is an implementation detail about what instrumentation is chosen to be used for the given deployment scenario.

Reply:
The health check leverages app insights directly via its `ITelemetryClient`, so we're sort of prescribing the instrumentation, right? But at least if I add the `services.AddApplicationInsightsTelemetry` in the same method as the health check, ensure all to the dependencies are added with the health check. If the user wants to override the instrumentation key, they can still call this method later to successfully update it.